### ISSUE TITLE
Mssql: several warnings with the addition of #5915

### ISSUE THREAD

**Issue by waader** [joomla/joomla-cms#10327]
#### Steps to reproduce the issue
Install current staging with mssql and testing data.

#### Actual result
Go to the frontend and observer several warnings:

Warning: explode() expects parameter 2 to be string, array given in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 376

Warning: array_merge() [function.array-merge]: Argument #1 is not an array in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 443

Warning: array_unique() expects parameter 1 to be array, null given in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 443

Warning: implode() [function.implode]: Invalid arguments passed in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 444

**Comment by alikon** [comment]
please check with the latest staing, thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10327">issues.joomla.org/joomla-cms/10327</a>.</sub>

**Comment by waader** [comment]
Same result.

**Comment by alikon** [comment]
@waader can you please update your info about errors with the latest staging <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10327">issues.joomla.org/joomla-cms/10327</a>.</sub>

**Comment by waader** [comment]
Nothing changed:

Warning: explode() expects parameter 2 to be string, array given in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 376

Warning: array_merge() [function.array-merge]: Argument #1 is not an array in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 443

Warning: array_unique() expects parameter 1 to be array, null given in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 443

Warning: implode() [function.implode]: Invalid arguments passed in C:\Program Files (x86)\Ampps\www\joomla35\libraries\joomla\database\query\sqlsrv.php on line 444

**Comment by andrepereiradasilva** [comment]
i think the problem come from this coomit https://github.com/joomla/joomla-cms/commit/55078e01f9af04abf7fe38c66d4909e0fe878000

**Comment by sovainfo** [comment]
@waader Please try : 
```
is_string($columns) && $columns = explode(',', str_replace(" ", "", $columns));
```
on line 376, it should make sure $columns is always an array.
Expect all warnings to disappear.

**Comment by waader** [comment]
@sovainfo: the notices have gone!

**Comment by sovainfo** [comment]
See https://github.com/joomla/joomla-cms/pull/10480
@waader Please mark the successful test

Request for closing issue.

**Comment by brianteeman** [comment]
Closed as we have a PR #10480<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10327">issues.joomla.org/joomla-cms/10327</a>.</sub>
